classes/local/hook_callbacks.php

```diff
@@ -0,0 +1,92 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+namespace tiny_elements\local;
+
+use core\hook\output\before_http_headers;
+
+/**
+ * Class containing the hook callbacks for tiny_elements.
+ *
+ * @package    tiny_elements
+ * @copyright  2024 ISB Bayern
+ * @author     Philipp Memmel
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class hook_callbacks {
+    /**
+     * Hook callback function for the before_http_headers hook.
+     *
+     * Used to add our custom stylesheet and js to the DOM.
+     *
+     * @param before_http_headers $beforehttpheadershook
+     */
+    public static function add_elements_data_to_dom(\core\hook\output\before_http_headers $beforehttpheadershook): void {
+        global $PAGE;
+
+        // Parameter to disable css delivery.
+        if (optional_param('tiny_elements_disable', false, PARAM_BOOL)) {
```

Comment:
What's this optional_param for? It does not occur anywhere else? Left over "debugging feature"?

Reply:
This should be kept for emergency situations when broken CSS / JS stops the page from working.